feat(sync): Add `close_replica` and enable subscriptions without active sync

Frando opened the pull request:
## Description

Based on #1491 

This adds two related features that were missing so far:

* Add `iroh_sync::store::close_replica` to remove a replica instance from the store-internal cache of open replicas. This is needed because otherwise it grows unbounded (for anchor nodes especially).
* Properly handle open replicas in the `LiveSync`, and through this allow to subscribe to replica events for replicas that are not in the set of actively syncing replicas. This was not possible before (you could only subscribe to replicas that are active for sync)
* Add test for subscribing without sync

## Notes & open questions

<!-- Any notes, remarks or open questions you have to make about the PR. -->

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [x] Tests if relevant.

dignifiedquire commented on iroh/src/sync_engine/live.rs:
```diff
@@ -506,8 +528,16 @@ impl<S: store::Store, B: baomap::Store> Actor<S, B> {
+        if !self.syncing_replicas.contains(namespace) {
+            None
+        } else {
+            self.replica_store.open_replica(namespace).ok().flatten()
```
should probably return the error or at least log it

dignifiedquire reviewed (approved):
lgtm, other than my small notes

Frando commented:
Thanks! Addressed the comments. As this is based on #1491 - Should I merge into that one now or wait for that to be reviewed, to keep reviewing easier?

b5 commented:
I just ran this up in iOS land, and it's fixed a bug I was experiencing with the node crashing on an attempt to load keys!

dignifiedquire commented:
feel free to pick your merge order, I don’t mind for review
